### PR TITLE
fix(parental-leave): Fix and refactor InReview screen steps and status

### PR DESCRIPTION
**Pull request by madebynoam**
* Removes the `FormMode` from the InReview form (since we can't currently know when VMLST fully approves the application).
* Refactors the InReview screen's steps to better handle dynamic cases (`isSelfEmployed` and `isRequestingRights`)

https://app.asana.com/0/1182378413629561/1200214178491351/f
https://app.asana.com/0/1182378413629561/1200418421137534/f

## Screenshots / Gifs
image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx [hunk 1]
```diff
@@ -52,7 +55,11 @@ const InReviewSteps: FC<FieldBaseProps> = ({
   refetch,
   errors,
 }) => {
-  const { isRequestingRights } = useApplicationAnswers(application)
+  const {
+    isSelfEmployed,
+    isRequestingRights,
+    usePersonalAllowanceFromSpouse,
```
Comment (baering) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:61]: Keep in mind that these are not boolean but `yes` / `no`. So it is not enough to check for `true` / `false` but you'll need to check `if (isSelfEmployed === YES) { ... }`

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx [hunk 2]
```diff
@@ -67,34 +74,42 @@ const InReviewSteps: FC<FieldBaseProps> = ({
...
+      title: formatMessage(parentalLeaveFormMessages.reviewScreen.deptTitle),
       description: formatMessage(
-        parentalLeaveFormMessages.reviewScreen.otherParentDesc,
+        parentalLeaveFormMessages.reviewScreen.deptDesc,
       ),
     },
-    {
+  ]
+
+  if (!isSelfEmployed) {
```
Comment (baering) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:85]: `if (isSelfEmployed === NO)`

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx [hunk 3]
```diff
@@ -67,34 +74,42 @@ const InReviewSteps: FC<FieldBaseProps> = ({
...
-    },
-  ]
+    })
+  }
+
+  if (requiresOtherParentApproval(application.answers)) {
+    const description =
+      isRequestingRights && usePersonalAllowanceFromSpouse
+        ? parentalLeaveFormMessages.reviewScreen.otherParentDescRequestingBoth
+        : isRequestingRights
```
Comment (baering) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:101]: `isRequestingRights === YES`

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx [hunk 4]
```diff
@@ -67,34 +74,42 @@ const InReviewSteps: FC<FieldBaseProps> = ({
...
-        parentalLeaveFormMessages.reviewScreen.deptDesc,
-      ),
-    },
-  ]
+    })
+  }
+
+  if (requiresOtherParentApproval(application.answers)) {
+    const description =
+      isRequestingRights && usePersonalAllowanceFromSpouse
```
Comment (baering) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:99]: `isRequestingRights === YES && usePersonalAllowanceFromSpouse === YES`

**Review by baering — CHANGES_REQUESTED** [review]
LGTM. One thing I noticed is that you're expecting some values to be `boolean` when they are in fact a custom `Boolean` type with the strings `'yes'`or `'no'` vs `true` or `false`.

**Comment by madebynoam** [comment]
Thanks for the review and heads up, should be fixed now.

**Review by baering — APPROVED** [review]

**Pull request merged**